### PR TITLE
Collapse RBAC into calico.yaml

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Build on top of this PR: https://github.com/projectcalico/calico/pull/2233

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
CC @2ffs2nns I think once this goes in, our Jenkins jobs might stop working since they'll expect an rbac file that no longer exists.

File: master/manifests/rbac-etcd-calico.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+layout: null
+---
+{% include {{page.version}}/manifests/rbac-etcd-calico.yaml %}
```
Comment (fasaxc) [inline master/manifests/rbac-etcd-calico.yaml]: What's with these new RBAC files (I must be missing something!)?

File: _includes/master/manifests/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  name: canal-calico
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: calico-node
+subjects:
+- kind: ServiceAccount
+  name: canal
+  namespace: kube-system
+{%- endif %}
```
Comment (emanic) [inline _includes/master/manifests/rbac.yaml]: Wow, looks so much better @caseydavenport 💯

File: master/getting-started/kubernetes/installation/integration.md [hunk 3]
```diff
@@ -204,11 +204,29 @@ tokens or certificates to present which identify it as the configured API user.
...
+
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/rbac-etcd-calico.yaml
+   ```
+
+- **etcd datastore with flannel networking**:
 
-[Click here to view the above yaml directly.](rbac.yaml)
+   ```
+   kubectl apply -f {{site.url}}/{{page.version}}/manifests/rbac-etcd-flannel.yaml
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/integration.md]: Maybe the above bullets need adjustment now though @caseydavenport

File: master/manifests/rbac-etcd-calico.yaml (hunk 1 above)
Comment (emanic) [reply master/manifests/rbac-etcd-calico.yaml]: These are there primarily to handle the integration use case. I think it is just a matter of adding the flags, as per @caseydavenport 's new changes. I can take a stab at it.

File: master/getting-started/kubernetes/installation/integration.md (hunk 3 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/integration.md]: Which ones?

**Comment by emanic** [comment]
🎉

**Pull request merged**